lxd/device/gpu_physical.go

```diff
@@ -365,16 +415,23 @@ func (d *gpuPhysical) postStop() error {
 			"last_state.pci.slot.name": "",
 			"last_state.pci.driver":    "",
 			"vgpu.uuid":                "",
+			"cdi":                      "",
 		})
 	}()
 
 	v := d.volatileGet()
 
 	if d.inst.Type() == instancetype.Container {
-		// Remove host files for this device.
-		err := unixDeviceDeleteFiles(d.state, d.inst.DevicesPath(), "unix", d.name, "")
-		if err != nil {
-			return fmt.Errorf("Failed to delete files for device '%s': %w", d.name, err)
+		// If we are in a CDI mode, the underlying postStop hooks for the various unix-char and disk devices
+		// are already registered when `stopWithCDISpec` is called.
+		//
+		// Else, we only have the unix-char device files to delete.
+		if v["cdi"] != "true" {
+			// Remove host files for this device.
+			err := unixDeviceDeleteFiles(d.state, d.inst.DevicesPath(), "unix", d.name, "")
+			if err != nil {
+				return fmt.Errorf("Failed to delete files for device '%s': %w", d.name, err)
```

Comment:
Should use %q rather than ``'%s'` for strings

Reply:
fixed